game.py

```diff
@@ -75,12 +83,50 @@ def check_events(self):
                 if event.key == pygame.K_RIGHT:
                     self.RIGHT_KEY = False
 
+    def asteroids_update(self):
+        for a in self.asteroids:  # Move the asteroids
+            a.move_asteroid()
+            if a.remove:  # Delete asteroid
+                self.asteroids.pop(self.asteroids.index(a))
+            else:
+                if self.collision_rocket(a, self.rocket):
+                    self.asteroids.pop(self.asteroids.index(a))
+                    self.rocket.life -= 1
+                    print("Rocket lives: ", self.rocket.life)
+                else:
+                    for p in self.projectiles:
+                        if self.collision_projectile(a, p):
+                            self.asteroids.pop(self.asteroids.index(a))
+                            self.projectiles.pop(self.projectiles.index(p))
+                            break
+
     def projectiles_update(self):
         for p in self.projectiles:  # Move the projectiles
             p.move_projectile()
             if p.remove:  # Delete projectiles
                 self.projectiles.pop(self.projectiles.index(p))
 
+    def collision_projectile(self, asteroid, projectile):
+        if asteroid.y + asteroid.height > projectile.y \
+                and asteroid.y < projectile.y + projectile.height \
+                and asteroid.x < projectile.x + projectile.width \
+                and asteroid.x + asteroid.width > projectile.x:
+            return True
+        else:
+            return False
+
+    def collision_rocket(self, asteroid, rocket):
```

Comment:
This should maybe go in the rocket class since it concerns the rocket itself.

Reply:
Same thing here indeed. I moved it to the `rocket` class